.github/workflows/publish.yml

```diff
@@ -1,84 +1,81 @@
+      - uses: actions/checkout@v2
+      - name: Get npm cache directory
+        id: npm-cache
+        run: |
+          echo "::set-output name=dir::$(npm config get cache)"
+      - uses: actions/cache@v2
+        with:
+          path: ${{ steps.npm-cache.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+      - run: npm ci
+      - run: npm run lint
+      - run: npm run check-circular
+  test:
+    strategy:
+      matrix:
+        node-version: [12.x, 14.x, 16.x]
+        os: [ubuntu-latest, windows-latest, macos-latest]
+    runs-on: ${{ matrix.os }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          submodules: recursive
+      - name: Get npm cache directory
+        if: runner.os != 'Windows'
+        id: npm-cache
+        run: |
+          echo "::set-output name=dir::$(npm config get cache)"
+      - uses: actions/cache@v2
+        if: runner.os != 'Windows' # Must be a bug on windows because its doing more harm than good. Try to cache/load almost a GB.
+        with:
+          path: ${{ steps.npm-cache.outputs.dir }}
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm ci
+      - run: npm run build
+      - run: npm test
   publish_cli:
     name: Publish CLI
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v2
-        with:
-          ref: 'master'
       - uses: actions/setup-node@v2
         with:
           node-version: '16'
       - run: npm install
       - run: npm run build
-      - name: Set NPM version and update README
-        id: npm-version-bump
-        run: |
-          echo "::set-output name=version::$(npm version ${GITHUB_REF/refs\/tags\//} --git-tag-version=false --allow-same-version)"
-          git config --global user.name "Github Actions"
-          git config --global user.email "bot@architect.io"
-          npm run readme
-          git add package.json package-lock.json README.md
-          git commit -m "Published CLI ${GITHUB_REF/refs\/tags\//}"
-          git push "https://${{ github.actor }}:${{ secrets.GITHUB_TOKEN }}@github.com/${{ github.repository }}.git"
+      - run: npm run readme
       - run: npm run pack
-      - name: Archive artifact
-        uses: actions/upload-artifact@v2
-        with:
-          name: dist
-          path: dist
       - name: npm publish
         run: |
           echo "//registry.npmjs.org/:_authToken=${{ secrets.NPM_PUBLISH_TOKEN }}" > .npmrc
-          npm publish
+          cross-env GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD) npm run semantic-release -- --dry-run
```

Comment:
Does cross-env exist in the image? Might make more sense to `npm install --save-dev cross-env` as part of the cli.

Then the package.json cmd would be:
`"semantic-release": "cross-env GIT_BRANCH=$(git rev-parse --abbrev-ref HEAD) ./node_modules/semantic-release/bin/semantic-release.js"`

Reply:
That's probably fine. Still my q applies. Is cross-env installed in the image for this job?